crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -100,6 +152,52 @@ impl Piece {
     pub fn as_mut(&mut self) -> PieceRefMut<'_> {
         PieceRefMut(&mut self.0)
     }
+
+    /// Split piece into underlying components.
+    pub fn split(&self) -> (RecordRef<'_>, WitnessRef<'_>) {
+        let (record, witness) = self.0.split_at(RECORD_SIZE as usize);
+        (
+            RecordRef(record),
+            WitnessRef(
+                witness
+                    .try_into()
+                    .expect("Witness withing a piece has correct size; qed"),
+            ),
+        )
+    }
+
+    /// Split piece into underlying mutable components.
+    pub fn split_mut(&mut self) -> (RecordRefMut<'_>, WitnessRefMut<'_>) {
+        let (record, witness) = self.0.split_at_mut(RECORD_SIZE as usize);
```

Comment:
I think we should replace `RECORD_SIZE` everywhere as an implicit parameter. `Archiver`, for example, accepts it as explicit parameter, while almost everywhere it is implicit.

Reply:
Yeah... I was going back and forth on this. We have some code in the codebase that is generic over record size and large part of archiver is in fact and can continue to work when they change on protocol level.

The issue is that `Piece` and related data structures rely globally on a fixed piece/record size and making them generic didn't look like a good idea due to the fact that it would infect and make generic over piece size ~30% of the codebase. I attempted it once and decided it is not worth it.

So I think we will be replacing it more going forward, but I also wanted to introduce smarter constants that can be converted into both `usize` and `u32`/`u64` without using `as`, such that we have static guarantees about integers fitting into one another.

There will be more PRs doing that, one of the upcoming PRs will stop compiling on 8-bit and 16-bit platforms with static assertions for instance, so we can introduce types and be sure that `u32 as usize` never looses any bits.

TL;DR: This is planned, but we're not there yet.